src/main/java/edu/project2/tripplanner/service/TripService.java

```diff
@@ -23,44 +23,57 @@
 
     private TripRepository tripRepository;
     private PlaceRepository placeRepository;
-    private CommentRepository commentRepository;
     private UserRepository userRepository;
 
 
     @Override
     public void deleteTripById(Long id) {
-     tripRepository.deleteTripById(id);
+        tripRepository.deleteTripById(id);
     }
 
     @Override
-    public ResponseEntity<Place> deletePlaceFromTrip(Long userId, Long tripId, Long placeId) {
-        tripRepository.findTripByIdAndUserId(userId,tripId).map(t->t.getPlaceSet().remove(placeRepository.findById(placeId).get()))
+    public Trip deletePlaceFromTrip(Long userId, Long tripId, Long placeId) {
+        Place place = placeRepository.findById(placeId).get();
+        return tripRepository.findTripByIdAndUserId(userId, tripId)
+                .map(t -> {
+                    t.removePlace(place);
+                    placeRepository.save(place);
+                    return tripRepository.save(t);
+                })
                 .orElseThrow(NotFoundExceptions::new);
-        return ResponseEntity.status(HttpStatus.OK)
-                .build();
     }
 
     @Override
-    public void addTrip(Trip trip) {
-    tripRepository.save(trip);
-//        tripRepository.addTrip(trip.getDepartureDay(),trip.getDayOfArrival(),trip.getTransport());
+    public ResponseEntity<Trip> addTrip(Long userId, Trip trip) {
+        userRepository.findById(userId).map(u -> {
+            trip.setDepartureDay(trip.getDepartureDay());
+            trip.setDayOfArrival(trip.getDayOfArrival());
+            Random random = new Random();
+            int i = random.nextInt(Transport.values().length);
+            trip.setTransports(Transport.values()[i]);
+            trip.setPrice((random.nextInt(1000)) + 100);
+            u.getTripList().add(trip);
+            trip.setUser(u);
+            return tripRepository.save(trip);
+        });
+        return ResponseEntity.status(HttpStatus.OK)
+                .build();
     }
 
     @Override
     public Trip editTrip(Long userId, Long tripId, Trip trip) {
         if (!userRepository.existsById(userId) | !tripRepository.existsById(tripId)) {
             throw new NotFoundExceptions();
         }
-        return tripRepository.findTripByIdAndUserId(userId,tripId).map(t ->{
```

Comment:
same as above. You should take a user from DB, if it doesn't exist, throw exception. If exist - execute logic